Neos.ContentRepositoryRegistry/Classes/Factory/ProjectionCatchUpTrigger/SubprocessProjectionCatchUpTrigger.php

```diff
@@ -29,15 +38,80 @@ public function triggerCatchUp(Projections $projections): void
     {
         // modelled after https://github.com/neos/Neos.EventSourcing/blob/master/Classes/EventPublisher/JobQueueEventPublisher.php#L103
         // and https://github.com/Flowpack/jobqueue-common/blob/master/Classes/Queue/FakeQueue.php
-        foreach ($projections as $projection) {
-            Scripts::executeCommandAsync(
-                'neos.contentrepositoryregistry:subprocessprojectioncatchup:catchup',
-                $this->flowSettings,
-                [
-                    'contentRepositoryIdentifier' => $this->contentRepositoryId->value,
-                    'projectionClassName' => get_class($projection)
-                ]
-            );
+        $queuedProjections = array_map(fn($projection) => $this->startCatchUpWithQueueing($projection), iterator_to_array($projections->getIterator()));
+        $queuedProjections = array_filter($queuedProjections);
+
+        $attempts = 0;
+        while (!empty($queuedProjections)) {
+            usleep(random_int(100, 25000) + ($attempts * $attempts * 10)); // 50000μs = 50ms
```

Comment:
Yes to both :)

I tried to make this more of an exponential backoff with an element of randomness to avoid any potential lockstep issues, will extract.